core/desugarLAttributes.ml

```diff
@@ -15,7 +15,7 @@ open SourceCode.WithPos
 
 let desugaring_error pos message =
   let open Errors in
-  raise (desugaring_error ~pos ~stage:DesugarLAttributes ~message)
+  desugaring_error ~pos ~stage:DesugarLAttributes ~message
```

Comment:
You raise a good point in that the exact raise location does not matter as much for user errors as for internal errors. 

That said, I don't think it hurts either way: the user won't see a stack trace by default. I've gone for uniformity, but if you would rather I factored out the `raise` for user errors, I'm happy to do so.

Reply:
I'd be all for uniformity. I complained because this seemed to me like a lack of uniformity, i.e. `desugaring_error` is a user error but unlike other such errors it does not include `raise` in its definition.

Reply:
Ah, I see! I missed that there are other such helper functions which do include raise in their definitions. Thanks for clarifying.

In that case, I will ensure that all of these exception generation functions do not `raise`, so everything is uniform.